sources/ActionView.cpp

```diff
@@ -24,17 +28,38 @@ ActionView::ActionView(const char* name, BMessage* action, const int32& flags)
 
 	RuleRunner::GetActions(fActions);
 
+	fFilePanels = new BFilePanel*[nActions];
+	fRefFilters = new TypedRefFilter*[nActions];
+	for (int32 i = 0; i < nActions; i++) {
+		fFilePanels[i] = NULL;
+		fRefFilters[i] = NULL;
+	}
+
 	fActionField = new BMenuField(NULL, ActionMenu());
+	fIconButton = new BButton(NULL, new BMessage(MSG_ACTION_PANEL));
 
 	fValueBox = new AutoTextControl("valuebox", NULL, NULL, new BMessage());
 	fValueBox->SetDivider(0);
 
+	size_t size;
+	const uint8* data = static_cast<const uint8*>(BApplication::AppResources()->
+							LoadResource(B_VECTOR_ICON_TYPE, 1, &size));
+
+	const float height = fValueBox->Bounds().Height() - 12;
+	fIcon = new BBitmap(BRect(0, 0, height, height), 0, B_RGBA32);
+
+	BIconUtils::GetVectorIcon(data, size, fIcon);
+	fIconButton->SetIcon(fIcon);
+
 	fAddRemoveButtons = new AddRemoveButtons(MSG_ADD_ACTION, MSG_REMOVE_ACTION,
-		this);
+		this, fValueBox->Bounds().Height());
 
 	BLayoutBuilder::Group<>(this, B_HORIZONTAL, B_USE_HALF_ITEM_SPACING)
 		.Add(fActionField, 0)
-		.Add(fValueBox)
+		.AddGroup(B_HORIZONTAL, 0)
+			.Add(fIconButton, 0)
+			.Add(fValueBox)
+			.End()
 		.Add(fAddRemoveButtons)
```

Comment:
What do you think about adding a `.AddStrut(B_USE_HALF_ITEM_SPACING)` before the +/- buttons?
Separates the buttons a little bit that aren't part of the actual rule.
Would have to be added to the test view too.
Also, a bit wider window would give a little more space to the text control.